node/src/proxy_server/mod.rs

```diff
@@ -2241,26 +2552,87 @@ mod tests {
                 .proxy_server(proxy_server_mock)
                 .neighborhood(neighborhood_mock)
                 .build();
-            // Get the add_route recipient so we can partially mock it...
-            let add_route_recipient = peer_actors.proxy_server.add_route;
+            // Get the dns_retry_result recipient so we can partially mock it...
+            let dns_retry_result_recipient = peer_actors.proxy_server.route_result_sub;
             peer_actors.proxy_server = ProxyServer::make_subs_from(&subject_addr);
-            peer_actors.proxy_server.add_route = add_route_recipient; //Partial mocking
+            peer_actors.proxy_server.route_result_sub = dns_retry_result_recipient; //Partial mocking
             subject_addr.try_send(BindMessage { peer_actors }).unwrap();
 
             subject_addr.try_send(msg_from_dispatcher).unwrap();
 
             system.run();
         });
 
-        let expected_add_route_message = AddRouteMessage {
+        let expected_dns_retry_result_message = RouteResultMessage {
             stream_key,
-            route: route_query_response.unwrap(),
+            result: Ok(route_query_response.unwrap()),
         };
 
         proxy_server_awaiter.await_message_count(1);
         let recording = proxy_server_recording_arc.lock().unwrap();
-        let record = recording.get_record::<AddRouteMessage>(0);
-        assert_eq!(record, &expected_add_route_message);
+        let message = recording.get_record::<RouteResultMessage>(0);
+        assert_eq!(message, &expected_dns_retry_result_message);
+    }
+
+    #[test]
+    fn proxy_server_sends_a_message_when_dns_retry_cannot_find_a_route() {
+        let test_name = "proxy_server_sends_a_message_when_dns_retry_cannot_find_a_route";
+        let cryptde = main_cryptde();
+        let http_request = b"GET /index.html HTTP/1.1\r\nHost: nowhere.com\r\n\r\n";
+        let (proxy_server_mock, _, proxy_server_recording_arc) = make_recorder();
+        let route_query_response = None;
+        let (neighborhood_mock, _, _) = make_recorder();
+        let neighborhood_mock = neighborhood_mock
+            .stop_on_message(type_id!(RouteResultMessage))
```

Comment:
You should delete this macro `type_id!`. The idea has always been that all the programmer needs can be done by `match_every_type_id!()`. It's designed to handle whatever, a single or multiple messages specified.

Reply:
This is something that I proposed to Syther that we should implement. I first request him to add this and now requesting him to remove it, will mean just adding more trouble in Syther's life.

Here are some reasons why I asked him for such a request.
1) The `type_id!` macro can also be used while debugging where a developer wants to just print the type id while testing to make sure, it's what they are expecting.
2) If the developer wants to do the above without the use of this macro, it's a bit hard to remember what exact code they must write to get the type id. You may look into the implementation of the macro, you'll also find this macro easier to use rather than the original code.
3) This macro was the first macro that Syther wrote and it'll be better if we keep it as it'll be more beneficial for everyone to remember his efforts.

Let me know if it's appropriate to keep it. 🙏